### PR TITLE
EAR-1737-Dynamic-Answers

### PR DESCRIPTION
**Pull request by oms09**
### What is the context of this PR?

> This PR allows Authors to be able to use an undefined number of checkbox answer labels to populate another set (subsequent) of radio answer labels as the respondent needs see previously selected answers in order to answer a follow on question. 

### How to review

1. Create a checkbox answer
2. Create a radio answer

> - [ ] Check dynamic option toggle switch appears for a radio answer 

4. Turn the dynamic option toggle switch on 

> - [ ] Check when the dynamic option toggle switch is on the dynamic answer content picker button displays and the label and description fields disappear.

5. Click the dynamic answer content picker button and select a checkbox answer in the modal and click confirm

> - [ ]  Check the correct checkbox answer is displayed in the content picker modal
> - [ ]  Check when the answer is clicked in the modal the background for the selected answer turns blue
> - [ ]  Check the correct checkbox answer is displayed in the  dynamic answer content picker button after clicking confirm

Test that it works correctly with multiple checkbox answers in the questionnaire - repeating the steps above

**Validation** 

- [ ] Check the validation message "Answer required" displays under the  dynamic answer content picker button when an answer hasn't been selected
- [ ]  Check the validation message "The selected answer has been deleted" displays under the  dynamic answer content picker button when the checkbox answer selected has been deleted 
- [ ] Check the validation message "Answer must be from a previous question" displays under the  dynamic answer content picker button when the checkbox answer selected has moved to a position after the current page

### What to do after everything is green

1. - [ ] Bring this branch up-to-date with Origin/Master
3. - [ ] If there are a lot of commits or some are not helpful to read, squash them down
4. - [ ] Click the **Merge** button on this pull request
6. - [ ] Does this change mean the **Capability examples** questionnaire should be updated?
7. - [ ] Move the Jira ticket for this task into the next stage of the process

### CODE REVIEW

**Review by farres1 — CHANGES_REQUESTED** [review]
These changes are looking really good!

A change I've noticed, currently the "What is a dynamic option" collapsible is not displayed when the dynamic option is toggle switch is off. On the prototype, when the dynamic option toggle switch is off, the collapsible is displayed.

**Review by farres1 — CHANGES_REQUESTED** [review]
Adding a list of changes discussed during a call:

- Add the collapsible when the toggle switch is off (from the prototype)
- Add the word “question” in the collapsible sentence “Note, if only one checkbox answer exists then the radio answer question is skipped.” (from  the prototype)
- Show the “Off On” labels on the toggle switch (from the prototype)
- Replace the `<p>` tags with styled components to improve consistency with other components
- Remove the `useCurrentPageId.mockImplementation(() => "1.1.2")` line in files edited where it doesn’t cause tests to break
- Add an underscore `_` in the variable `DYNAMICANSWER` to make it `DYNAMIC_ANSWER`, to follow the codebase's UPPER_CASE format and improve readability

**Review by farres1 — APPROVED** [review]

**Review by farres1 — APPROVED** [review]

File: eq-author/src/App/page/Design/answers/MultipleChoiceAnswer/Option.test.js [hunk 1]
```diff
@@ -5,19 +5,30 @@ import { StatelessOption } from "./Option";
 import { CHECKBOX, RADIO } from "constants/answer-types";
 import { merge } from "lodash";
 import { useMutation } from "@apollo/react-hooks";
+import { useCurrentPageId } from "components/RouterContext";
```
Comment (farres1) [inline eq-author/src/App/page/Design/answers/MultipleChoiceAnswer/Option.test.js:8]:
```suggestion

```
This line can now be removed as useCurrentPageId is no longer user

File: eq-author/src/App/page/Design/answers/MultipleChoiceAnswer/index.test.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { render } from "enzyme";
 import { CHECKBOX, RADIO } from "constants/answer-types";
 import createMockStore from "tests/utils/createMockStore";
 import suppressConsoleMessage from "tests/utils/supressConsol";
+import { useCurrentPageId } from "components/RouterContext";
```
Comment (farres1) [inline eq-author/src/App/page/Design/answers/MultipleChoiceAnswer/index.test.js:8]:
```suggestion

```
This line can now be removed as useCurrentPageId is no longer user

**Review by farres1 — CHANGES_REQUESTED** [review]

**Review by farres1 — APPROVED** [review]

**Review by ashtuck — APPROVED** [review]

**Review by ashtuck — APPROVED** [review]

**Review by farres1 — APPROVED** [review]

**Review by ashtuck — APPROVED** [review]

**Pull request merged**